include/ttmlir/Dialect/TTNN/IR/TTNNOpsAttrs.td

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTMLIR_TTMLIR_DIALECT_TTNN_TTNNOPSATTRS_TD
+#define TTMLIR_TTMLIR_DIALECT_TTNN_TTNNOPSATTRS_TD
+
+include "mlir/IR/AttrTypeBase.td"
+include "mlir/IR/EnumAttr.td"
+include "mlir/IR/BuiltinTypeInterfaces.td"
+include "mlir/IR/CommonTypeConstraints.td"
+include "ttmlir/Dialect/TTNN/IR/TTNNBase.td"
+include "ttmlir/Dialect/TTNN/IR/TTNNOpsEnums.td"
+
+//===----------------------------------------------------------------------===//
+// TTNN attr definitions
+//===----------------------------------------------------------------------===//
+
+class TTNN_Attr<string name, string attrMnemonic, list<Trait> traits = [],
+                   string baseCppClass = "::mlir::Attribute">
+    : AttrDef<TTNN_Dialect, name, traits, baseCppClass> {
+  let mnemonic = attrMnemonic;
+  let attrName = "ttnn." # attrMnemonic;
+}
+
+def TTNN_CoreRangeAttr : TTNN_Attr<"CoreRange", "core_range"> {
+  let summary = "TTNN grid attribute";
+  let description = [{
+    TTNN grid attribute
+  }];
+
+  let parameters = (ins ArrayRefParameter<"int64_t">:$offset,
+                        ArrayRefParameter<"int64_t">:$size);
+  let assemblyFormat = "`<` custom<DimensionList>($offset) `,` custom<DimensionList>($size) `>`";
+
+  let extraClassDeclaration = [{
+      static CoreRangeAttr get(::mlir::MLIRContext *context, ::mlir::tt::GridAttr grid, SmallVector<int64_t> offset = {0, 0})
+      {
+        assert(grid.getShape().size() == 2 && "Grid shape must be 2D for now");
+        return CoreRangeAttr::get(context, {0, 0}, grid.getShape());
+      }
+  }];
+}
+
+def TTNN_CoreRangeArrayAttr : TypedArrayAttrBase<TTNN_CoreRangeAttr, "">;
```

Comment:
I'm actually not sure! All the CoreRange* attrs here show as additions because I separated Types and Attrs, and these were previously in `TTNNOpsTypes.td` file.